gobblin-runtime/src/main/java/org/apache/gobblin/scheduler/JobScheduler.java

```diff
@@ -412,20 +411,19 @@ protected void logNewlyScheduledJob(JobDetail job, Trigger trigger) {
 
   /**
    * Unschedule and delete a job.
+   * Returns true is job was scheduled and is successfully unscheduled, false otherwise
    *
    * @param jobName Job name
    * @throws JobException when there is anything wrong unschedule the job
    */
-  public void unscheduleJob(String jobName)
+  public boolean unscheduleJob(String jobName)
```

Comment:
I think for the sake of swallowing exceptions it's trading changing a class function signature to move the log to info. Is it possible instead just to either ignore if the scheduledJobs does not contain the key? Seems that's the RC of the noisy logs where it would delete a job that does not exist in the cache.

Reply:
yes, RC is that it throws exception in GobblinServiceJobScheduler when the job does not exist in the map.
If we ignore when the scheduledJobs does not contain the key and return normally (return void, instead of throwing exception like it is doing now), then the next line `this.orchestrator.remove(deletedSpec, headers)` will get executed, which I want to avoid.
Basically, right now we are throwing exceptions to avoid running `this.orchestrator.remove(deletedSpec, headers)`, the only other way to tell the caller not to run further statements is return a flag.

Reply:
oh, btw problem is that the job does not exist in GobblinServiceJobScheduler::scheduledFlowSpecs
not in JobScheduler::scheduledJobs